vnext/Shared/HermesSamplingProfiler.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include <chrono>
+
+#include <hermes/hermes.h>
+#include <winrt/Windows.Storage.h>
+#include "Unicode.h"
+
+#include "HermesSamplingProfiler.h"
+
+namespace Microsoft::ReactNative {
+
+namespace {
+std::future<std::string> getTraceFilePath() {
+  auto hermesFolder = (co_await winrt::Windows::Storage::ApplicationData::Current().LocalFolder().CreateFolderAsync(
+                           L"Hermes", winrt::Windows::Storage::CreationCollisionOption::OpenIfExists))
+                          .Path();
+  auto now = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch())
+                 .count();
+  std::ostringstream os;
+  os << Microsoft::Common::Unicode::Utf16ToUtf8(hermesFolder.c_str(), hermesFolder.size()) << "\\cpu_" << +now
+     << ".cpuprofile";
+  co_return os.str();
+}
+} // namespace
+
+bool HermesSamplingProfiler::s_isStarted = false;
+std::string HermesSamplingProfiler::s_lastTraceFilePath;
+
+std::string HermesSamplingProfiler::GetLastTraceFilePath() {
+  return s_lastTraceFilePath;
+}
+
+winrt::fire_and_forget HermesSamplingProfiler::Start() {
+  s_isStarted = true;
+  facebook::hermes::HermesRuntime::enableSamplingProfiler();
```

Comment:
Is the hermes code idempotent? If not should we add a check if in case double clicks get through before we can disable the button?

Reply:
Thanks ! 
Done.. Please note that i assume these gets called on UI thread always ... hence don't bother synchronizing the condition.